internal/pkg/manifest/transform.go

```diff
@@ -18,58 +21,284 @@ var basicKinds = []reflect.Kind{
-		return transformStringSliceOrString(reflect.TypeOf(CommandOverride{}))
+	if typ.String() == "map[string]manifest.Volume" {
+		return transformMapStringToVolume
 	}
 
-	if typ == reflect.TypeOf(Alias{}) {
-		return transformStringSliceOrString(reflect.TypeOf(Alias{}))
+	if transform := transformBasic(typ); transform != nil {
+		return transform
 	}
 
-	if typ.String() == "map[string]manifest.Volume" {
-		return transformMapStringToVolume()
+	return nil
+}
+
+func (t workloadTransformer) transformCompositeTypes(typ reflect.Type) func(dst, src reflect.Value) error {
+	var et exclusiveTransformer
+	switch typ {
+	case reflect.TypeOf(EntryPointOverride{}), reflect.TypeOf(CommandOverride{}), reflect.TypeOf(Alias{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstStruct := dst.Convert(reflect.TypeOf(stringSliceOrString{})).Interface().(stringSliceOrString)
+			srcStruct := src.Convert(reflect.TypeOf(stringSliceOrString{})).Interface().(stringSliceOrString)
+			if err := mergo.Merge(&dstStruct, srcStruct, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstStruct).Convert(typ))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"String"}, []string{"StringSlice"})
+		}
+	case reflect.TypeOf(PlatformArgsOrString{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstPlatformArgsOrString := dst.Interface().(PlatformArgsOrString)
+			srcPlatformArgsOrString := src.Interface().(PlatformArgsOrString)
+			if err := mergo.Merge(&dstPlatformArgsOrString, srcPlatformArgsOrString, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstPlatformArgsOrString))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"PlatformString"}, []string{"PlatformArgs"})
+		}
+	case reflect.TypeOf(HealthCheckArgsOrString{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstHC := dst.Interface().(HealthCheckArgsOrString)
+			srcHC := src.Interface().(HealthCheckArgsOrString)
+			if err := mergo.Merge(&dstHC, srcHC, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstHC))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"HealthCheckPath"}, []string{"HealthCheckArgs"})
+		}
+	case reflect.TypeOf(Count{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstC := dst.Interface().(Count)
+			srcC := src.Interface().(Count)
+			if err := mergo.Merge(&dstC, srcC, opts(countTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstC))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"Value"}, []string{"AdvancedCount"})
```

Comment:
This particular signature I think makes the codebase fragile to refactor. Imagine renaming the field `AdvancedCount` to `CountArgs`. Now the implementor has to remember that there is a string named "AdvancedCount" here as well and modify the method. I bet it's unlikely that we will remember as the IDE won't pick up that this field needs to be updated.

We should think of some tenets for this project 💭 For example:
- **Seamless integrations with IDEs.** Refactoring a field name or type should not break existing merge behavior.
- **Transparent**. It should be clear when we need to add a new transformer.

If the suggestion of calling each individual transformer one-by-one works, can we instead move the exclusive field logic there?
```go
func (t countTransformer) transformExclusiveTypes(typ reflect.Type) func(dst, src reflect.Value) error {
	if typ != reflect.TypeOf(AdvancedCount{}) {
		return nil
	}
	// merge like before ...
	// But then check for fields like this: dstC.Spot != nil and dstC.Range != nil 
	// So don't generalize.	
}
```

Reply:
done!